mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/search/AllSearchOverride.java

```diff
@@ -65,6 +67,7 @@ private boolean isEmpty(SearchQuery searchQuery) {
 
     @Override
     public Flux<MessageUid> search(MailboxSession session, Mailbox mailbox, SearchQuery searchQuery) {
-        return dao.listAllMessageUid((PostgresMailboxId) mailbox.getMailboxId());
+        return Mono.fromCallable(() -> new PostgresMailboxMessageDAO(executorFactory.create(session.getUser().getDomainPart())))
```

Comment:
I don't think that's what I meant by more elegant... Honestly if we use search overrides is it really normal to recreate a new PostgresMailboxMessageDAO object every time we do a search? I think in terms of perf that should create quickly issues when you have thousands users searching conccurently, no?

What's the error message without this "fix"?

Reply:
> Honestly if we use search overrides is it really normal to recreate a new PostgresMailboxMessageDAO object every time we do a search? I think in terms of perf that should create quickly issues when you have thousands users searching conccurently, no?

It is not only a "search override" story, we get it when we use RLS for the mailbox, we create a new DAO for each mailboxSession request (not only  `PostgresMailboxMessageDAO`)
// I don't know, it is my concern too...

> I don't think that's what I meant by more elegant

Can you give more detail?

Reply:
Not sure what to think TBH... Can we wait to see your results with perf tests first? I'm curious